drivers/vl53l1x/vl53l1x.c

```diff
@@ -0,0 +1,1292 @@
+    uint8_t ambient_count_rate_mcps_sd0_hi     : 8;   /* fixed point 9.7 */
+    uint8_t ambient_count_rate_mcps_sd0_lo     : 8;   /* fixed point 9.7 */
+    uint8_t sigma_sd0_hi : 8;                         /* fixed point 14.2 */
+    uint8_t sigma_sd0_lo : 8;                         /* fixed point 14.2 */
+    uint8_t phase_sd0_hi : 8;                         /* fixed point 5.11 */
+    uint8_t phase_sd0_lo : 8;                         /* fixed point 5.11 */
+    uint8_t final_crosstalk_corrected_range_mm_sd0_hi : 8;
+    uint8_t final_crosstalk_corrected_range_mm_sd0_lo : 8;
+    uint8_t peak_signal_count_rate_crosstalk_corrected_mcps_sd0_hi : 8; /* fixed point 9.7 */
+    uint8_t peak_signal_count_rate_crosstalk_corrected_mcps_sd0_lo : 8; /* fixed point 9.7 */
+} _vl53l1x_raw_data_t;
+
+static int _update(const vl53l1x_t *dev, uint16_t index, uint8_t mask, uint8_t data);
+static int _read(const vl53l1x_t *dev, uint16_t index, uint8_t *data, uint32_t len);
+static int _read_byte(const vl53l1x_t *dev, uint16_t index, uint8_t *data);
+static int _read_word(const vl53l1x_t *dev, uint16_t index, uint16_t *data);
+static int _write(const vl53l1x_t *dev, uint16_t index, const uint8_t *data, uint32_t len);
+static int _write_byte(const vl53l1x_t *dev, uint16_t index, uint8_t data);
+static int _write_word(const vl53l1x_t *dev, uint16_t index, uint16_t data);
+#if !MODULE_VL53L1X_BASIC
+static int _write_dword(const vl53l1x_t *dev, uint16_t index, uint32_t data);
+#endif /* !MODULE_VL53L1X_BASIC */
+
+static int _start(const vl53l1x_t *dev);
+static int _update_dss(const vl53l1x_t *dev, _vl53l1x_raw_data_t* raw_data);
+
+#if !MODULE_VL53L1X_BASIC
+static int _setup_manual_calibration(const vl53l1x_t *dev);
+static uint16_t _enc_timeout(uint32_t clks);
+static uint32_t _dec_timeout(uint16_t val);
+static uint32_t _timeout_clk_to_us(uint32_t clks, uint32_t period_us);
+static uint32_t _timeout_us_to_clk(uint32_t us, uint32_t period_us);
+static uint32_t _calc_macro_period(const vl53l1x_t *dev, uint8_t vcsel_period);
+#endif /* !MODULE_VL53L1X_BASIC */
+
+/* Copyrighted POLOLU / ST API code begin */
+static const uint32_t _timing_guard = 4528;  /* used in timing budget calculations */
+static const uint16_t _target_rate = 0x0A00; /* used in DSS calculations */
+/* Copyrighted POLOLU / ST API code end */
+
+#endif /* MODULE_VL53L1X_ST_API */
+
+int vl53l1x_init(vl53l1x_t *dev, const vl53l1x_params_t *params)
+{
+    /* some parameter sanity checks */
+    ASSERT_PARAM(dev != NULL);
+    ASSERT_PARAM(params != NULL);
+    #if !MODULE_VL53L1X_BASIC
+    ASSERT_PARAM(params->budget <= 1000000);
+    ASSERT_PARAM(params->period >= params->budget/1000 + 4);
+    #endif /* !MODULE_VL53L1X_BASIC */
+    DEBUG_DEV("params=%p", dev, params);
+
+    /* init sensor data structure */
+    dev->params = *params;
+    dev->int_init = false;
+
+    #if MODULE_VL53L1X_ST_API
+    dev->dev.I2cDevAddr = VL53L1X_I2C_ADDRESS;
```

Comment:
The VL53L1X has a fixed default I2C address. Although it can be changed with an ST-API function, I don't know how this would work if there are multiple devices in the system that all have the same default slave address to begin with.